src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -309,9 +320,9 @@ public void RecoverTxSenders(Block block)
             }
         }
 
-        public void RecoverTxSender(Transaction tx, long blockNumber)
+        public void RecoverTxSender(Transaction tx, long? blockNumber)
         {
```

Comment:
only do that if needed - very expensive operation - check here if sender is null